### ISSUE TITLE
【FR】[Hygon/BW200] gemma-3-1b-it: linear/mm ops cause 25.9% accuracy degradation on GPQA Diamond

### ISSUE THREAD

**Issue by ckxud** [FlagOpen/FlagGems#4837]
### Description

During accuracy evaluation on model **google/gemma-3-1b-it** with FlagGems enabled on Hygon BW200, GPQA Diamond benchmark shows 25.93% relative accuracy degradation compared to NVIDIA baseline. Operator-level diagnosis identified `linear` and `mm` as the primary contributors, but a residual gap remains after disabling them.

### Environment

| Item | Value |
|------|-------|
| Hardware | Hygon BW200 (DCU) ×8 |
| TP Size | 1 |
| PyTorch | 2.10.0 |
| FlagGems | 5.4.0dev |
| vLLM | 0.20.2 |
| vllm-plugin-FL | 0.2.0 |
| DTK | 26.04-DCC2602-0317 |
| Model | google/gemma-3-1b-it (~1.9B params) |

### Steps to Reproduce

1. Start vLLM service with FlagGems enabled (35 operators loaded):
   ```bash
   VLLM_PLUGINS=fl USE_FLAGGEMS=1 vllm serve google/gemma-3-1b-it \
     --port 8001 --tensor-parallel-size 1 --trust-remote-code
   ```
2. Run GPQA Diamond accuracy evaluation (50 questions)
3. Compare score against NVIDIA reference baseline (27.0%)

### Expected Behavior

FlagGems-enabled accuracy should be within 5% relative deviation from NVIDIA baseline (27.0%).

### Actual Behavior

| Metric | Value |
|--------|-------|
| NV Baseline | 27.0% |
| V2 (FlagGems, all ops) | 20.0% |
| Relative degradation | **25.93%** (threshold: 5%) |

After operator-level accuracy tuning (4 rounds), best configuration:

| Round | Disabled Ops | Score | Rel Drop | Notes |
|-------|-------------|-------|----------|-------|
| Initial | (none) | 20.0% | 25.9% | All 35 ops enabled |
| R1 | linear, mm | 24.0% | 11.1% | +4 points improvement |
| R2 | linear, mm, full, zeros | 24.0% | 11.1% | No additional improvement |
| R3 | linear, mm, full, zeros, add, cos, div, reciprocal, sin, sub | 20.0% | 25.9% | Regressed — math ops needed |
| R4 | linear, mm, cumsum, argmax, sum, softmax | 20.0% | 25.9% | Regressed — reduce ops needed |

**Best achievable**: 24.0% with `linear` + `mm` disabled (still 11.1% > 5% threshold).

### Affected Operators

- **`linear`** (`flag_gems.ops.linear.linear`) — primary contributor
- **`mm`** (`flag_gems.ops.mm.mm_out`) — primary contributor

Disabling both recovers 4 percentage points (20% → 24%), but a residual 11.1% gap remains that is not attributable to any single FlagGems operator group.

### FlagGems Integration

**Code path**: `/opt/vllm/vllm/model_executor/models/minicpmo.py`
**Enable call**: `flag_gems.only_enable(record=False, include=FLAG_GEMS_CONFIG)`
**Loaded operators**: 35 (via `/tmp/flaggems_enable_oplist.txt`)

### Analysis

1. `linear` and `mm` are compute-heavy operators — precision loss in matrix multiplication accumulates through transformer layers
2. The residual gap (11.1%) after disabling linear/mm suggests additional precision issues in the combined operator pipeline, possibly from interaction effects
3. This is specific to Hygon BW200 platform; NVIDIA baseline achieves 27.0% on the same model

### Suggested Investigation

- Profile `linear` and `mm` operator implementations for numerical precision on Hygon DCU
- Check FP16/BF16 intermediate accumulation behavior in Triton kernels
- Compare kernel outputs against PyTorch reference implementations with identical inputs
- Investigate if the residual gap relates to attention_backend dispatch or combined operator interactions
